extension/app/ts/utils/typescript.ts

```diff
@@ -36,3 +36,8 @@ export function assertsOneOf<T, U extends T>(item: T, array: readonly U[], messa
 export type DistributedPick<T, K extends string> = T extends unknown ? { [P in K & keyof T]: T[P] } : never
 export type DistributedOmit<T, K extends keyof T> = T extends unknown ? Omit<T, K> : never
 export type DistributiveOmit<T extends object, K extends keyof T> = T extends unknown ? Omit<T, K> : never
+
+export function asObject(object: unknown) : Object {
+	if (typeof object !== 'object' || object === null) throw new Error('Not object')
+	return object
+}
```

Comment:
6 suggestions in one:
1. `object` is a keyword, so I recommend not using it as a variable name as it can confuse some tools sometimes.
2. No space after parameters before return type `:`.
3. Change this into a type asserting function instead of something that returns itself.
4. Check to see if it is an array as well.
5. Include the object type in the error.
6. Make both a type checking and a type asserting function, as you may need both eventually.
```suggestion
export function isObject(maybe: unknown): maybe is Object {
	return typeof maybe === 'object' && maybe !== null && !Array.isArray(maybe)
}
export function assertIsObject(maybe: unknown): asserts maybe is Object {
	if (!isObject(maybe)) throw new Error('Expected object but got ${typeof maybe}')
}
```
Usage example:
```ts
declare const apple: unknown

apple // type: unknown
assertIsObject(apple)
apple // type: Object

declare const banana: unknown
if (isObject(banana)) {
	banana // type: Object
} else {
	banana // type: unknown
}
```